src/builder/analyzer.rs

```
@@ -803,16 +876,47 @@ impl AnalyzerContext {
                 let (struct_mapping, fields_schema) = analyze_struct_mapping(&op.input, op_scope)?;
                 let has_auto_uuid_field = op.auto_uuid_field.is_some();
                 let fingerprinter = Fingerprinter::default().with(&fields_schema)?;
+                let input_field_names: Vec<FieldName> =
+                    fields_schema.iter().map(|f| f.name.clone()).collect();
+                let collector_ref = add_collector(
+                    &op.scope_name,
+                    op.collector_name.clone(),
+                    CollectorSchema::from_fields(fields_schema, op.auto_uuid_field.clone()),
+                    op_scope,
+                )?;
+                // Get the merged collector schema after adding
+                let collector_schema: Arc<CollectorSchema> = {
+                    let scope = find_scope(&op.scope_name, op_scope)?.1;
+                    let states = scope.states.lock().unwrap();
+                    let collector = states.collectors.get(&op.collector_name).unwrap();
+                    collector.schema.clone()
+                };
+
+                // Pre-compute field index mappings for efficient evaluation
+                let field_name_to_index: HashMap<&FieldName, usize> = collector_schema
+                    .fields
+                    .iter()
+                    .enumerate()
+                    .map(|(i, f)| (&f.name, i))
+                    .collect();
+                let field_index_mapping: Vec<usize> = input_field_names
+                    .iter()
+                    .map(|field_name| {
+                        field_name_to_index
+                            .get(field_name)
+                            .copied()
+                            .unwrap_or(usize::MAX)
```

If not found in this case, it's a bug. So instead of `unwrap_or(usize::MAX)`, we should raise an error.

use something like:

```rs
ok_or_else(|| anyhow!("field `{}` not found in merged collector schema"))?;
```